### ISSUE TITLE
compiler warnings after upgrade

### ISSUE THREAD

**Issue by jgabry** [stan-dev/rstan#372]
#### Summary:
@bgoodri Any idea why after upgrading to 2.13.2 (installed from CRAN from source) I'm getting a bunch of unexpected output during comilation, seemingly related to __StanHeaders__ and __BH__?

#### Current Output:
```r
library("rstan")
fit <- stan_demo("eight_schools")
```

```r
> J <- 8

> y <- c(28,  8, -3,  7, -1,  1, 18, 12)

> sigma <- c(15, 10, 16, 11,  9, 11, 10, 18)

> tau <- 25

In file included from file1a4350c474ee.cpp:8:
In file included from /Library/Frameworks/R.framework/Versions/3.3/Resources/library/StanHeaders/include/src/stan/model/model_header.hpp:4:
In file included from /Library/Frameworks/R.framework/Versions/3.3/Resources/library/StanHeaders/include/stan/math.hpp:4:
In file included from /Library/Frameworks/R.framework/Versions/3.3/Resources/library/StanHeaders/include/stan/math/rev/mat.hpp:4:
In file included from /Library/Frameworks/R.framework/Versions/3.3/Resources/library/StanHeaders/include/stan/math/rev/core.hpp:12:
In file included from /Library/Frameworks/R.framework/Versions/3.3/Resources/library/StanHeaders/include/stan/math/rev/core/gevv_vvv_vari.hpp:5:
In file included from /Library/Frameworks/R.framework/Versions/3.3/Resources/library/StanHeaders/include/stan/math/rev/core/var.hpp:7:
In file included from /Library/Frameworks/R.framework/Versions/3.3/Resources/library/BH/include/boost/math/tools/config.hpp:13:
In file included from /Library/Frameworks/R.framework/Versions/3.3/Resources/library/BH/include/boost/config.hpp:39:
/Library/Frameworks/R.framework/Versions/3.3/Resources/library/BH/include/boost/config/compiler/clang.hpp:196:11: warning: 'BOOST_NO_CXX11_RVALUE_REFERENCES' macro redefined [-Wmacro-redefined]
#  define BOOST_NO_CXX11_RVALUE_REFERENCES
          ^
<command line>:6:9: note: previous definition is here
#define BOOST_NO_CXX11_RVALUE_REFERENCES 1
        ^
1 warning generated.

SAMPLING FOR MODEL 'eight_schools' NOW (CHAIN 1).
```
Then it starts sampling ok. 

#### Expected Output:
Same as above but without all the junk before it starts sampling.

#### RStan Version:
`2.13.2`

#### R Version:
`3.3.2`

#### Operating System:
`OS X 10.12.1`

**Comment by bgoodri** [comment]
Yeah, we had to add that `-DBOOST_NO_CXX11_RVALUE_REFERENCES`  to get around a Boost bug. I guess you just have to shut it up by adding -Wno-macro-redefined to CXXFLAGS in ~/.R/Makevars.

**Comment by jgabry** [comment]
Thanks, that worked. Is this something that should go in the installation
instructions for the time being (until rstan is released for first time
after Boost bug is fixed)?

**Comment by bgoodri** [comment]
We should probably mention in specifically in warnings.html.

**Comment by mdbauer** [comment]
When I add this option to my Makevars I get the following compiler warning:
`cc1plus.exe: warning: unrecognized command line option "-Wno-macro-redefined"`

**Comment by luisdamiano** [comment]
Similarly, I'm getting this warning (see last line):

```
In file included from C:/Users/Bebop/R/win-library/3.4/BH/include/boost/config.hpp:39:0,
                 from C:/Users/Bebop/R/win-library/3.4/BH/include/boost/math/tools/config.hpp:13,
                 from C:/Users/Bebop/R/win-library/3.4/StanHeaders/include/stan/math/rev/core/var.hpp:7,
                 from C:/Users/Bebop/R/win-library/3.4/StanHeaders/include/stan/math/rev/core/gevv_vvv_vari.hpp:5,
                 from C:/Users/Bebop/R/win-library/3.4/StanHeaders/include/stan/math/rev/core.hpp:12,
                 from C:/Users/Bebop/R/win-library/3.4/StanHeaders/include/stan/math/rev/mat.hpp:4,
                 from C:/Users/Bebop/R/win-library/3.4/StanHeaders/include/stan/math.hpp:4,
                 from C:/Users/Bebop/R/win-library/3.4/StanHeaders/include/src/stan/model/model_header.hpp:4,
                 from file23f075087533.cpp:8:
C:/Users/Bebop/R/win-library/3.4/BH/include/boost/config/compiler/gcc.hpp:186:0: warning: "BOOST_NO_CXX11_RVALUE_REFERENCES" redefined
 #  define BOOST_NO_CXX11_RVALUE_REFERENCES
 ^
<command-line>:0:0: note: this is the location of the previous definition
cc1plus.exe: warning: unrecognized command line option "-Wno-macro-redefined"
```

Even though I'm using these settings in my Makevars:

`CXXFLAGS=-O3 -Wno-unused-variable -Wno-unused-function -Wno-macro-redefined`

Session info:

```
 setting  value                       
 version  R version 3.4.1 (2017-06-30)
 system   x86_64, mingw32             
 ui       RStudio (1.1.447)           
 language (EN)                        
 collate  Spanish_Argentina.1252      
 tz       America/Buenos_Aires        
 date     2018-06-10
```

Packages:

```
 package      * version   date       source        
 BH             1.65.0-1  2017-08-24 CRAN (R 3.4.1)
 colorspace     1.3-2     2016-12-14 CRAN (R 3.4.1)
 dichromat      2.0-0     2013-01-24 CRAN (R 3.4.1)
 digest         0.6.12    2017-01-27 CRAN (R 3.4.1)
 ggplot2      * 2.2.1     2016-12-30 CRAN (R 3.4.1)
 graphics     * 3.4.1     2017-06-30 local         
 grDevices    * 3.4.1     2017-06-30 local         
 grid           3.4.1     2017-06-30 local         
 gridExtra      2.3       2017-09-09 CRAN (R 3.4.1)
 gtable         0.2.0     2016-02-26 CRAN (R 3.4.1)
 inline         0.3.14    2015-04-13 CRAN (R 3.4.1)
 labeling       0.3       2014-08-23 CRAN (R 3.4.1)
 lattice        0.20-35   2017-03-25 CRAN (R 3.4.1)
 lazyeval       0.2.0     2016-06-12 CRAN (R 3.4.1)
 magrittr       1.5       2014-11-22 CRAN (R 3.4.1)
 MASS           7.3-47    2017-02-26 CRAN (R 3.4.1)
 Matrix         1.2-10    2017-05-03 CRAN (R 3.4.1)
 methods      * 3.4.1     2017-06-30 local         
 munsell        0.4.3     2016-02-13 CRAN (R 3.4.1)
 plyr           1.8.4     2016-06-08 CRAN (R 3.4.1)
 R6             2.2.2     2017-06-17 CRAN (R 3.4.1)
 RColorBrewer   1.1-2     2014-12-07 CRAN (R 3.4.1)
 Rcpp         * 0.12.17   2018-05-18 CRAN (R 3.4.4)
 RcppEigen      0.3.3.3.0 2017-05-01 CRAN (R 3.4.1)
 reshape2       1.4.2     2016-10-22 CRAN (R 3.4.1)
 rlang          0.1.2     2017-08-09 CRAN (R 3.4.1)
 rstan        * 2.17.3    2018-01-20 CRAN (R 3.4.4)
 scales         0.5.0     2017-08-24 CRAN (R 3.4.1)
 StanHeaders  * 2.17.2    2018-01-20 CRAN (R 3.4.4)
 stats        * 3.4.1     2017-06-30 local         
 stats4         3.4.1     2017-06-30 local         
 stringi        1.1.5     2017-04-07 CRAN (R 3.4.1)
 stringr        1.2.0     2017-02-18 CRAN (R 3.4.1)
 tibble         1.3.4     2017-08-22 CRAN (R 3.4.1)
 tools          3.4.1     2017-06-30 local         
 utils        * 3.4.1     2017-06-30 local         
 viridisLite    0.2.0     2017-03-24 CRAN (R 3.4.1)
```
